### PR TITLE
Feat/create concert

### PR DESCRIPTION
**Pull request by Nishantchavda36**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,27 +3,27 @@ import { BrowserRouter, Routes, Route } from 'react-router-dom';
...
-import Signup from './pages/signup/Signup';
 import Lyrics from './pages/lyrics/Lyrics';
 import Concerts from './pages/concerts/Concerts';
 import Concert from './pages/concert/Concert';
 import CreateConcert from './pages/createConcert/CreateConcert';
+import { useAuthContext } from './hooks/useAuthContext';
 
 function App() {
+  const { user } = useAuthContext();
+  console.log(user);
```
Comment (hridaysur) [inline src/App.js]: Please remove this console.log if not required.

File: src/hooks/useAuthContext.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { useContext } from 'react';
+import { AuthContext } from '../contexts/AuthContext';
+
+export const useAuthContext = () => {
+  const context = useContext(AuthContext);
+
+  if (!context) {
+    throw Error('useAuthContext must be used inside an AuthContextProvider');
```
Comment (hridaysur) [inline src/hooks/useAuthContext.js:8]: Who will catch this error if thrown? How we will handle this?

File: src/hooks/useLogin.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { useAuthContext } from './useAuthContext';
+import { LOGIN } from '../contexts/AuthContext';
+
+const googleProvider = new GoogleAuthProvider();
+
+export const useLogin = () => {
+  const { dispatch } = useAuthContext();
+
+  const login = async () => {
+    console.log('clicked');
```
Comment (hridaysur) [inline src/hooks/useLogin.js]: please remove this console.log if not required.

File: src/hooks/useLogin.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    try {
+      const result = await signInWithPopup(auth, googleProvider);
+      // const credential = GoogleAuthProvider.credentialFromResult(result);
+      // const token = credential.accessToken;
```
Comment (hridaysur) [inline src/hooks/useLogin.js]: Can we please remove this comments? If this variables are require ltr we can use create them at that time.

File: src/hooks/useLogin.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        email,
+        photoURL,
+        uid,
+      };
+      dispatch({ type: LOGIN, payload: user });
+    } catch (error) {
+      // const errorCode = error.code;
+      // const errorMessage = error.message;
+      // const email = error.customData.email;
+      // const credential = GoogleAuthProvider.credentialFromError(error);
```
Comment (hridaysur) [inline src/hooks/useLogin.js]: also this comments

File: src/pages/createConcert/CreateConcert.js [hunk 6]
```diff
@@ -1,8 +1,108 @@
...
+  const [image, setImage] = useState('');
+
+  // Handle form submission
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (title && venue && date && image) {
+      console.log(title);
+      console.log(venue);
+      console.log(date);
+      console.log(image);
```
Comment (hridaysur) [inline src/pages/createConcert/CreateConcert.js:23]: console.log

**Review by hridaysur — CHANGES_REQUESTED** [review]

File: src/hooks/useAuthContext.js (hunk 2 above)
Comment (13prashant) [reply src/hooks/useAuthContext.js:8]: It's for development mode only!

File: src/hooks/useLogin.js (hunk 4 above)
Comment (13prashant) [reply src/hooks/useLogin.js]: It's still under process. Figuring out when and where to use them.

**Pull request merged**